libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -418,6 +418,9 @@ - (SFSyncState*)syncUpWithTarget:(SFSyncUpTarget *)target
                          options:(SFSyncOptions *)options
                         soupName:(NSString *)soupName
                      updateBlock:(SFSyncSyncManagerUpdateBlock)updateBlock {
+    if (target == nil) {
```

Comment:
This should fix hybrid smartsync tests: https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1901